### PR TITLE
[MOSIP-25064] Updated helm to enable prometheus scraping for regproc stage groupes

### PR DESCRIPTION
**Pull request by akilalakshmanan**

### CODE REVIEW

File: charts/regproc-group2/templates/service.yaml [hunk 1]
```diff
@@ -22,7 +22,14 @@ spec:
   loadBalancerIP: {{ .Values.service.loadBalancerIP }}
   {{- end }}
   ports:
-    - port: {{ .Values.service.port }}
-      protocol: TCP 
-      targetPort: {{ .Values.springServicePort }}
+    - name: securezone
```
Comment (syedsalman3753) [inline charts/regproc-group2/templates/service.yaml:25]: do not hardcode values.

please pass it via values.yaml in range format as used for other services.yaml

**Review by syedsalman3753 — CHANGES_REQUESTED** [review]

File: charts/regproc-group2/templates/service.yaml [hunk 2]
```diff
@@ -22,7 +22,9 @@ spec:
   loadBalancerIP: {{ .Values.service.loadBalancerIP }}
   {{- end }}
   ports:
-    - port: {{ .Values.service.port }}
-      protocol: TCP 
-      targetPort: {{ .Values.springServicePort }}
+  {{- range $ep := .Values.metrics.endpointPathList }}
+  - name: {{ $ep.port_name }}
```
Comment (syedsalman3753) [inline charts/regproc-group2/templates/service.yaml:26]: target port is missing

File: charts/regproc-group2/values.yaml [hunk 3]
```diff
@@ -376,10 +376,13 @@ metrics:
 
   endpointPathList:
   - endpointPath: '/registrationprocessor/v1/securezone/actuator/prometheus'
+    port_name: securezone
     port: 8090
```
Comment (syedsalman3753) [inline charts/regproc-group2/values.yaml:380]: add target ports

File: charts/regproc-group3/templates/service.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  {{- end }}
+  {{ if eq .Values.service.type "LoadBalancer" }}
+  loadBalancerSourceRanges: {{ .Values.service.loadBalancerSourceRanges }}
+  {{ end }}
+  {{- if (and (eq .Values.service.type "LoadBalancer") (not (empty .Values.service.loadBalancerIP))) }}
+  loadBalancerIP: {{ .Values.service.loadBalancerIP }}
+  {{- end }}
+  ports:
+  {{- range $ep := .Values.metrics.endpointPathList }}
+  - name: {{ $ep.port_name }}
```
Comment (syedsalman3753) [inline charts/regproc-group3/templates/service.yaml:26]: target port is missing

File: charts/regproc-group3/values.yaml [hunk 5]
```diff
@@ -375,10 +375,13 @@ metrics:
 
   endpointPathList:
     - endpointPath: '/registrationprocessor/v1/biodedupe/actuator/prometheus'
+      port_name: biodedupe
       port: 9096
     - endpointPath: '/registrationprocessor/v1/abishandler/actuator/prometheus'
+      port_name: abishandler
```
Comment (syedsalman3753) [inline charts/regproc-group3/values.yaml:381]: add target port as well

File: charts/regproc-group4/templates/service.yaml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  {{- end }}
+  {{ if eq .Values.service.type "LoadBalancer" }}
+  loadBalancerSourceRanges: {{ .Values.service.loadBalancerSourceRanges }}
+  {{ end }}
+  {{- if (and (eq .Values.service.type "LoadBalancer") (not (empty .Values.service.loadBalancerIP))) }}
+  loadBalancerIP: {{ .Values.service.loadBalancerIP }}
+  {{- end }}
+  ports:
+  {{- range $ep := .Values.metrics.endpointPathList }}
+  - name: {{ $ep.port_name }}
```
Comment (syedsalman3753) [inline charts/regproc-group4/templates/service.yaml:26]: The target port is missing

Add the target ports for all server yamls and endpointList

**Review by syedsalman3753 — CHANGES_REQUESTED** [review]

**Review by syedsalman3753 — APPROVED** [review]

**Review by ckm007 — APPROVED** [review]

**Pull request merged**